pubsub/iterator.go

```diff
@@ -486,18 +474,17 @@ func (it *messageIterator) sendModAck(m map[string]bool, deadline time.Duration)
 				recordStat(it.ctx, ModAckTimeoutCount, 1)
 				return nil
 			default:
-				if err == nil {
-					return nil
-				}
 				// This addresses an error where `context deadline exceeded` errors
 				// not captured by the previous case causes fatal errors.
 				// See https://github.com/googleapis/google-cloud-go/issues/3060
-				if strings.Contains(err.Error(), "context deadline exceeded") {
+				if err != nil && strings.Contains(err.Error(), "context deadline exceeded") {
```

Comment:
Does the RPC error include an ErrorInfo? We should really be using that rather than relying on the error message.

Reply:
So this was a strange error that wasn't coming from the server (as far as I could tell) so I'm not entirely sure if it has `ErrorInfo`.